rpc/src/transaction_status_service.rs

```diff
@@ -20,8 +20,13 @@ use {
     },
 };
 
+// Used when draining and shutting down TSS in unit tests.
+const TSS_TEST_QUIESCE_NUM_RETRIES: usize = 100;
+const TSS_TEST_QUIESCE_SLEEP_TIME_MS: u64 = 50;
+
 pub struct TransactionStatusService {
     thread_hdl: JoinHandle<()>,
+    transaction_status_receiver: Arc<Receiver<TransactionStatusMessage>>,
```

Comment:
Bit of an odd corner case, but consider the following sequence:
- Producer pumps a bunch of data into channel
- Exit flag set, producer stops and sender is dropped
- Sender gets dropped while stuff still in channel

In this scenario, I'm not positive if we'll be able to continue reading from the channel OR if we'll get get disconnected error. If disconnected, then somewhat defeats the purpose
https://docs.rs/crossbeam/latest/crossbeam/channel/enum.TryRecvError.html

Do you happen to know the behavior here off-hand ? If not, we can probably dig around in code and/or test it our real quick. Some code as simple as this would seemingly do the trick
```rust
let (sender, receiver) = unbounded!();
sender.send(3).unwrap();
drop(sender);
let maybe3 = receiver.try_recv().unwrap();
```

Reply:
Nice ! Think we're good to go as-is then